### ISSUE TITLE
Fatal: --quiet and --verbose cannot be specified at the same time

### ISSUE THREAD

**Issue by whereisaaron** [restic/restic#1796]
## Output of `restic version`

`restic 0.9.0 compiled with go1.10.2 on linux/amd64`

## How did you run restic exactly?

```
            command:
            - "/usr/local/bin/restic"
            args:
            - "backup"
            - "--quiet"
            - "--verbose"
            - "--verbose"
            - "--hostname=foo"
            - "--exclude-caches"
            - "--exclude-if-present=auto.cnf"
            - "--tag=job"
            - "/vol/foo/"
```

## What backend/server/service did you use to store the repository?

B2

## Expected behavior

This is a batch run, and so the pre-scan and progress display is a waste of time. However, I want a detailed record of the actions the backup tool in the log. Hence `--quiet --verbose --verbose`.

- I expected `--quiet` to disable the pre-scan
- I expected `--verbose --verbose` to enable logging files or file-chucks included in the backup

## Actual behavior

I got errors: "Fatal: --quiet and --verbose cannot be specified at the same time"

## Steps to reproduce the behavior

See 'How did you run restic exactly?' above.

## Do you have any idea what may have caused this?

It doesn't seem like the functionality overloaded onto these settings overlaps. A verbose log of actions during the backup doesn't need the pre-scan does it?

## Do you have an idea how to solve the issue?

1) Remove the prohibition over using `--quiet` and `--verbose` together. Although it is weird in English to use them at the same time, the functionality gated by them makes sense to use at the same time.

2) Alternatively, don't overload settings, make `--no-prescan` or `--no-progress` its own option, and make `--quiet` and `--verbose` a convenient bundle of more fine-grained settings.

## Did restic help you or made you happy in any way?

Looking forward to giving 0.9.0 a good test!

**Comment by fd0** [comment]
Hey, thanks for the report. I think you've read about `--quiet` in the changelog, that was a bit unfortunate... It was a workaround for the old version of the archiver code, which did not start reading files until the scan was done. So we decided to just skip the scan if `--quiet` is passed.

With the new archiver code, scanning and the backup process is run in parallel, so restic won't wait until the scanner is done, but directly start reading and saving files.

So, the changelog entry is unfortunate, but in general I don't think this is a bug.

What are your thoughts on it?

**Comment by whereisaaron** [comment]
I'm confused. This is the change log for 0.9.0, this behavior it is listed as an enhancement specifically for 0.9.0, but you say it doesn't apply to 0.9.0? If the change is no longer in the code or redundant, shouldn't it be removed from the change log? This functionality is not a change from 0.8.3 -> 0.9.0 since it apparently isn't in either version?

---------
Enhancement #1676: Improve backup speed: Skip initial scan phase in quiet mode

We've improved the backup speed when the quiet flag (-q or --quiet) is set by skipping the initial scan which gathers information for displaying the progress bar and the ETA estimation.

#1160 #1676
---------

**Comment by whereisaaron** [comment]
This is still a bug, as I think I still can't enable the output options you need for batch tasks? I don't want the interactive, multi-line status display and `--quiet` is apparently the only way to disable that? But I do want the log entries for the files processed, which apparently is only enabled by `--verbose` or `--verbose --verbose`? But I can't use both at once, which (in my head) doesn't make much sense 😄 

https://github.com/restic/restic/releases/tag/v0.9.0
> While we were at it, the status display and notification system was completely rewritten. By default, restic now shows which files are currently read (unless --quiet is specified) in a multi-line status display.
>
> The backup command also gained a new option: --verbose. It can be specified once (which prints a bit more detail what restic is doing) or twice (which prints a line for each file/directory restic encountered, together with some statistics).

The current `--quiet` option, when used on its own, disables all output unless there is an error - is that right? That is an ideal option for cron jobs where any stderr/stdout output is considered something that needs to be emailed/notified. Good option for `crontab` 👍 

However other batch systems (e.g. Kubernetes Jobs/CronJobs, Octopus Deploy) consider the exit value or the (lack of) stderr output as the indication of success/failure. Output from stdout is considered log content and stored for later review.

How can users disable the interactive multi-line status display, but still enable the logging?

**Comment by fd0** [comment]
> If the change is no longer in the code or redundant, shouldn't it be removed from the change log?

Indeed, I'll remove it. It was present in an intermediate version that did not get released at all because #1494 was merged and replaced the code.

Usually, we're trying to document the user-facing changes in the code directly in the PRs, so they don't get lost (and I don't get to spend time writing a changelog when I'm releasing a new version). We ran into a special case here :)

> This is still a bug, as I think I still can't enable the output options you need for batch tasks? I don't want the interactive, multi-line status display and --quiet is apparently the only way to disable that? But I do want the log entries for the files processed, which apparently is only enabled by --verbose or --verbose --verbose? But I can't use both at once, which (in my head) doesn't make much sense ￼

It must be quite confusing for you as a user, I'm sorry about that. The options `--quiet` and `--verbose` control how much output is printed in the "log" part of restic's output. The status lines are not controlled with either option, when a terminal is available (only when restic is run interactively), the status lines are shown. In all other cases, the status lines are not shown, so you don't get a huge email when running restic via cron.

Currently, there's no way to explicitly enable/disable the status lines, it feels to me the automatic detection (is a terminal there or not?) works really well. What are your thoughts on this?

Maybe we should document this in the manual somewhere...

> The current --quiet option, when used on its own, disables all output unless there is an error - is that right?

Correct.

> How can users disable the interactive multi-line status display, but still enable the logging?

Easy: Pass `--verbose` to restic (or `-v -v`), the rest is automatic. If you run it interactively, the status lines will be shown, otherwise not.

If you'd like to hide the status lines and still run interactively, you can force `stdout` to not be a terminal by piping it to `cat`, e.g. `restic -v -v backup [...] | cat`.

**Comment by whereisaaron** [comment]
Thanks @fd0, getting clearer!

Automatic suppression base on lack of terminal is great default behavior. Thanks for letting me know, and yes please add to the manual.

You could still consider a `--no-progress` or `--noninteractive` or similar option to force the no-terminal-present behavior. This is a pretty common option on tools that get used in both modes (e.g. apt, composer). And more elegant that piping through another process to fake the non-terminal situation. If people use that option when the automatic detection would have handled it, no harm done I think? Not essential/required of course.

Ok, once more I change the options on these restic jobs! My git log is full of 'add option x', 'remove option x, add y', 'add option x back' 😄

**Comment by fd0** [comment]
So, I'd like to reopen this issue to track adding some documentation to the manual. I'll think about ading the CLI option.

**Comment by cgonzalez** [comment]
Looking forward for a `--noninteractive` option! I'm making backups for several Windows servers. Restic is executed through a centralized task management platform called Rundeck (https://www.rundeck.com/open-source) using the WinRM plugin. It seems that Restic's automatic suppression based on lack of terminal is not working when using Rundeck and therefore we get a huge report file after every execution.

**Comment by fd0** [comment]
Ok, thanks for the report.
